program/src/processor.rs

```diff
@@ -723,18 +723,14 @@ pub fn process_withdraw_inactive_stake(
         stake_observed_total = (stake_observed_total + account_balance)?;
     }
 
-    if stake_observed_total < validator.entry.effective_stake_balance() {
-        // Solana has no slashing at the time of writing, and only Solido can
-        // withdraw from these accounts, so we should not observe a decrease in
-        // balance.
-        msg!(
-            "Observed balance of {} is less than tracked balance of {}.",
-            stake_observed_total,
-            validator.entry.effective_stake_balance()
-        );
-        msg!("This should not happen, aborting ...");
-        return Err(LidoError::ValidatorBalanceDecreased.into());
-    }
+    // Solana has no slashing at the time of writing, and only Solido can
+    // withdraw from these accounts, so we should not observe a decrease in
+    // balance.
+    Validator::observe_balance(
+        stake_observed_total,
+        validator.entry.effective_stake_balance(),
+        "Stake",
+    )?;
 
     // We tracked in `stake_accounts_balance` what we put in there ourselves, so
     // the excess is a donation by some joker.
```

Comment:
This comment is no longer true, now that we no longer work with 100%-commission accounts, the excess also represents staking rewards that are paid out by Solana.

Reply:
But still there could be a joker who transfers to stake account and it is considered as rewards by solido. So excess is a sum of rewards + joker transfer. When donation arrives to stake account it is inactive, but when rewards are payed of they become active immediately. That's the only difference. I will fix the comment